### PR TITLE
bsp: imx-atf: imx8mp: enable sip call for secondary boot

### PR DESCRIPTION
**Pull request by vanmaegima**
This fixes rollback support in imx8mp-lpddr4-evk.

Signed-off-by: Vanessa Maegima <vanessa.maegima@foundries.io>

### CODE REVIEW

**Review by MrCry0 — APPROVED** [review]
LGTM

File: meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-atf/imx-atf/0001-plat-imx8mp-SiP-call-for-secondary-boot.patch [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
++		if (x2)
++			mmio_setbits_32(IMX_SRC_BASE + SRC_GPR10_OFFSET,
++					SRC_GPR10_PERSIST_SECONDARY_BOOT);
++		else
++			mmio_clrbits_32(IMX_SRC_BASE + SRC_GPR10_OFFSET,
++					SRC_GPR10_PERSIST_SECONDARY_BOOT);
++		break;
++	case IMX_SIP_SRC_GET_SECONDARY_BOOT:
++		val = mmio_read_32(IMX_SRC_BASE + SRC_GPR10_OFFSET);
++		return !!(val & SRC_GPR10_PERSIST_SECONDARY_BOOT);
```
Comment (caiotpereira) [inline meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-atf/imx-atf/0001-plat-imx8mp-SiP-call-for-secondary-boot.patch:31]: is the "!!" necessary?

Comment (vanmaegima) [reply meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-atf/imx-atf/0001-plat-imx8mp-SiP-call-for-secondary-boot.patch:31]: We're just forcing the answer to be '1' or '0'
We are doing the same for all the patches for this support: https://github.com/foundriesio/meta-lmp/blob/main/meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-atf/imx-atf/0004-plat-imx8mq-SiP-call-for-secondary-boot.patch#L30

Comment (caiotpereira) [reply meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-atf/imx-atf/0001-plat-imx8mp-SiP-call-for-secondary-boot.patch:31]: ok, Thanks !

**Review by Tim-Anderson — APPROVED** [review]
LGTM

**Pull request merged**